scene/gui/scroll_container.cpp

```diff
@@ -360,6 +360,46 @@ void ScrollContainer::_reposition_children() {
 		size.x -= v_scroll->get_minimum_size().x;
 	}
 
+	float viewport_w = size.x;
+	float viewport_h = size.y;
+
+	float total_content_w = 0.0f;
+	float total_content_h = 0.0f;
+	for (int ci = 0; ci < get_child_count(); ci++) {
+		Control *cc = as_sortable_control(get_child(ci));
+		if (!cc || cc == h_scroll || cc == v_scroll || cc == focus_panel) {
+			continue;
+		}
+		Size2 ms = cc->get_combined_minimum_size();
+		float child_w = ms.x;
+		float child_h = ms.y;
+		if (cc->get_h_size_flags().has_flag(SIZE_EXPAND)) {
+			child_w = MAX(viewport_w, ms.x);
+		}
+		if (cc->get_v_size_flags().has_flag(SIZE_EXPAND)) {
+			child_h = MAX(viewport_h, ms.y);
+		}
+		total_content_w = MAX(total_content_w, child_w);
+		total_content_h += child_h;
+	}
+
+	float max_scroll_x = total_content_w > viewport_w ? (total_content_w - viewport_w) : 0.0f;
+	float max_scroll_y = total_content_h > viewport_h ? (total_content_h - viewport_h) : 0.0f;
+	float scroll_x = h_scroll ? CLAMP(h_scroll->get_value(), 0.0f, max_scroll_x) : 0.0f;
+	float scroll_y = v_scroll ? CLAMP(v_scroll->get_value(), 0.0f, max_scroll_y) : 0.0f;
+
+	float group_v_offset = 0.0f;
+	if (max_scroll_y <= 0.0f && total_content_h < viewport_h) {
+		int align_v = static_cast<int>(vertical_content_align);
+		float extra = viewport_h - total_content_h;
+		if (align_v == CONTENT_V_ALIGN_CENTER) {
+			group_v_offset = floorf(extra * 0.5f);
```

Comment:
```suggestion
			group_v_offset = Math::floor(extra * 0.5f);
```
Please use the built-in methods, for all these